gems/smithy-client/lib/smithy-client/block_io.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    # IO object for response targets.
+    class BlockIO
+      # @param [Hash] headers (nil)
+      # @param [Proc] block
+      def initialize(headers = nil, &block)
+        @headers = headers
```

Comment:
Why do we have headers on the blockIO?  Isn't it just intended to replace the body?

Reply:
It's given as a block argument. I suppose you would want to know this for content type or length?